### PR TITLE
[Doc] Document and Tutorial Structure Reorg for v0.3.1

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR reorgnizes the overall documentation and tutorial structure of GraphStorm based on the new structure design. As the distributed graph processing documents are still under structure, this PR leaves the distributed documents as it is, and modifies the rest.
Here comes the preview pages: https://james4graphstorm.readthedocs.io/en/james_reorg_doc4v031/index.html

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -58,47 +58,42 @@ Welcome to the GraphStorm Documentation and Tutorials
...
+For beginners, please first start with the :ref:`Setup GraphStorm with pip Packages<setup_pip>`. This tutorial covers how to set up a GraphStorm Standalone mode environment for quick start. 
 
-Once successfully set up the GraphStorm Docker running environment,
+Once successfully set up the GraphStorm running environment,
 
-- follow the :ref:`GraphStorm Standalone Mode Quick-Start Tutorial<quick-start-standalone>` to run examples using GraphStorm built-in data and models, hence getting familiar with GraphStorm's usage of training and inference.
-- follow the :ref:`Use Your Own Graph Data Tutorial<use-own-data>` to prepare your own graph data for using GraphStorm.
-- read the :ref:`GraphStorm Training and Inference Configurations<configurations-run>` to learn the various configurations provided by GraphStorm that can help to achieve the best performance.
+- follow the :ref:`GraphStorm Standalone Mode Quick-Start Tutorial<quick-start-standalone>` to use GraphStorm Command Line Interfaces (CLIs) to run exampless based on GraphStorm built-in data and models, hence getting familiar with GraphStorm CLIs for training and inference.
```
Comment (classicsong) [inline docs/source/index.rst:70]:
```suggestion
- follow the :ref:`GraphStorm Standalone Mode Quick-Start Tutorial<quick-start-standalone>` to use GraphStorm Command Line Interfaces (CLIs) to run examples based on GraphStorm built-in data and models, hence getting familiar with GraphStorm CLIs for training and inference.
```

File: docs/source/api/notebooks/index.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.. _programming-examples:
+
+API Programming Examples
+=========================
+
+Besides the Command Line Interfaces (CLIs), GraphStorm releases a set of Application Programming Interfaces (APIs) for users to build their own Graph Machine Learning (GML) components and pipelines. To help users to better use these APIs, we provide a set of Jupyter notebooks as examples. Users can download these notebooks and related python files, and then run them locally in the GraphStorm standalone mode.
```
[the remark this answers is not in the document]
Comment (thvasilo) [reply docs/source/api/notebooks/index.rst:6]: Yeah I guess we can just write models here?

[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/api/notebooks/index.rst:6]: Yes, it should be models. Updated.

File: docs/source/api/notebooks/index.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+These notebooks all use the same ACM data as discussed in the :ref:`User Your Own Data Tutorial<use-own-data>`. Users create the required dataset and explore details of the it by running the **Notebook 0: Data Prepare** notebook.
+
+The **Notebook 1** provides an example that demonstrate how to reproduce the node classification training and inference pipeline that is identical as the :ref:`Step 3: Launch training and inference scripts on your own graphs<launch_training_oyog>` does. While the **Notebook 2** provides another example that use the same dataset, but conduct a link prediction training and inference pipeline. 
+
+These two examples utilize GraphStorm APIs, such as ``GraphStorm``, ``GSgnnDataset``, ``GSgnnNodeDataLoader``, ``GSgnnNodePredictionTrainer``, and ``GSgnnNodePredictionInferrer``, to form the training and infernece pipeline. In terms of the GNN models, users can refer to the `demo_model.py <https://github.com/awslabs/graphstorm/blob/main/docs/source/notebooks/demo_models.py>`_ file in which all models are created by using GraphStorm APIs.
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/api/notebooks/index.rst:12]: Updated.

File: docs/source/api/references/index.rst [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.. _api-reference:
+
+API Reference
+================
+
+The released APIs include the necessary components that can build GraphStorm training and inference pipelines. For examples about , please refer to the :ref:`API Programming Examples<programming-examples>`.
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/api/references/index.rst]: Updated with the completed sentence.

**Review by classicsong — APPROVED** [review]
LGTM

**Pull request merged**